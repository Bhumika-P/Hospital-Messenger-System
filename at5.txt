add_user(1,"Sheree")
add_user(2,"Daniel")
add_user(3,"Edward")
add_group(1,"Friends")
add_group(2,"Neighbours")
register_user(1,1)
register_user(2,1)
register_user(1,2)
register_user(3,2)
send_message(1,2,"Hi , How are you doing?")
list_new_messages(1)
list_old_messages(1)
send_message(1,1,"I like pizza!")
list_new_messages(1)
list_old_messages(1)
send_message(2,1," WOOO , I like this")
set_message_preview(25)
list_new_messages(1)
list_old_messages(1)
read_message(1,1)
list_new_messages(1)
list_old_messages(1)
read_message(1,2)
list_new_messages(1)
list_old_messages(1)
read_message(2,2)
list_new_messages(2)
list_old_messages(2)
read_message(3,2)
list_new_messages(3)
list_old_messages(3)
read_message(1,3)
list_new_messages(1)
list_old_messages(1)
delete_message(2,1)
delete_message(2,2)
delete_message(2,3)    
list_old_messages(2)
list_new_messages(3)
