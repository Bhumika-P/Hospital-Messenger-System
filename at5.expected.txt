  0:  OK

->add_user(1,"Sheree")
  1:  OK
  Users:
      1->Sheree
  Groups:
  Registrations:
  All messages:
  New messages:
  Old/read messages:

->add_user(2,"Daniel")
  2:  OK
  Users:
      1->Sheree
      2->Daniel
  Groups:
  Registrations:
  All messages:
  New messages:
  Old/read messages:

->add_user(3,"Edward")
  3:  OK
  Users:
      1->Sheree
      2->Daniel
      3->Edward
  Groups:
  Registrations:
  All messages:
  New messages:
  Old/read messages:

->add_group(1,"Friends")
  4:  OK
  Users:
      1->Sheree
      2->Daniel
      3->Edward
  Groups:
      1->Friends
  Registrations:
  All messages:
  New messages:
  Old/read messages:

->add_group(2,"Neighbours")
  5:  OK
  Users:
      1->Sheree
      2->Daniel
      3->Edward
  Groups:
      1->Friends
      2->Neighbours
  Registrations:
  All messages:
  New messages:
  Old/read messages:

->register_user(1,1)
  6:  OK
  Users:
      1->Sheree
      2->Daniel
      3->Edward
  Groups:
      1->Friends
      2->Neighbours
  Registrations:
      [1, Sheree]->{1->Friends}
  All messages:
  New messages:
  Old/read messages:

->register_user(2,1)
  7:  OK
  Users:
      1->Sheree
      2->Daniel
      3->Edward
  Groups:
      1->Friends
      2->Neighbours
  Registrations:
      [1, Sheree]->{1->Friends}
      [2, Daniel]->{1->Friends}
  All messages:
  New messages:
  Old/read messages:

->register_user(1,2)
  8:  OK
  Users:
      1->Sheree
      2->Daniel
      3->Edward
  Groups:
      1->Friends
      2->Neighbours
  Registrations:
      [1, Sheree]->{1->Friends, 2->Neighbours}
      [2, Daniel]->{1->Friends}
  All messages:
  New messages:
  Old/read messages:

->register_user(3,2)
  9:  OK
  Users:
      1->Sheree
      2->Daniel
      3->Edward
  Groups:
      1->Friends
      2->Neighbours
  Registrations:
      [1, Sheree]->{1->Friends, 2->Neighbours}
      [2, Daniel]->{1->Friends}
      [3, Edward]->{2->Neighbours}
  All messages:
  New messages:
  Old/read messages:

->send_message(1,2,"Hi , How are you doing?")
  10:  OK
  Users:
      1->Sheree
      2->Daniel
      3->Edward
  Groups:
      1->Friends
      2->Neighbours
  Registrations:
      [1, Sheree]->{1->Friends, 2->Neighbours}
      [2, Daniel]->{1->Friends}
      [3, Edward]->{2->Neighbours}
  All messages:
      1->Hi , How are yo ...
  New messages:
      [3, Edward]->{1}
  Old/read messages:

->list_new_messages(1)
  11:  OK
  There are no new messages for this user.

->list_old_messages(1)
  12:  OK
  There are no old messages for this user.

->send_message(1,1,"I like pizza!")
  13:  OK
  Users:
      1->Sheree
      2->Daniel
      3->Edward
  Groups:
      1->Friends
      2->Neighbours
  Registrations:
      [1, Sheree]->{1->Friends, 2->Neighbours}
      [2, Daniel]->{1->Friends}
      [3, Edward]->{2->Neighbours}
  All messages:
      1->Hi , How are yo ...
      2->I like pizza!
  New messages:
      [2, Daniel]->{2}
      [3, Edward]->{1}
  Old/read messages:

->list_new_messages(1)
  14:  OK
  There are no new messages for this user.

->list_old_messages(1)
  15:  OK
  There are no old messages for this user.

->send_message(2,1," WOOO , I like this")
  16:  OK
  Users:
      1->Sheree
      2->Daniel
      3->Edward
  Groups:
      1->Friends
      2->Neighbours
  Registrations:
      [1, Sheree]->{1->Friends, 2->Neighbours}
      [2, Daniel]->{1->Friends}
      [3, Edward]->{2->Neighbours}
  All messages:
      1->Hi , How are yo ...
      2->I like pizza!
      3-> WOOO , I like  ...
  New messages:
      [1, Sheree]->{3}
      [2, Daniel]->{2}
      [3, Edward]->{1}
  Old/read messages:

->set_message_preview(25)
  17:  OK
  Users:
      1->Sheree
      2->Daniel
      3->Edward
  Groups:
      1->Friends
      2->Neighbours
  Registrations:
      [1, Sheree]->{1->Friends, 2->Neighbours}
      [2, Daniel]->{1->Friends}
      [3, Edward]->{2->Neighbours}
  All messages:
      1->Hi , How are you doing?
      2->I like pizza!
      3-> WOOO , I like this
  New messages:
      [1, Sheree]->{3}
      [2, Daniel]->{2}
      [3, Edward]->{1}
  Old/read messages:

->list_new_messages(1)
  18:  OK
  New/unread messages for user [1, Sheree]:
      3-> WOOO , I like this

->list_old_messages(1)
  19:  OK
  There are no old messages for this user.

->read_message(1,1)
  20:  ERROR 
  Message has already been read. See `list_old_messages'.

->list_new_messages(1)
  21:  OK
  New/unread messages for user [1, Sheree]:
      3-> WOOO , I like this

->list_old_messages(1)
  22:  OK
  There are no old messages for this user.

->read_message(1,2)
  23:  ERROR 
  Message has already been read. See `list_old_messages'.

->list_new_messages(1)
  24:  OK
  New/unread messages for user [1, Sheree]:
      3-> WOOO , I like this

->list_old_messages(1)
  25:  OK
  There are no old messages for this user.

->read_message(2,2)
  26:  OK
  Message for user [2, Daniel]: [2, "I like pizza!"]
  Users:
      1->Sheree
      2->Daniel
      3->Edward
  Groups:
      1->Friends
      2->Neighbours
  Registrations:
      [1, Sheree]->{1->Friends, 2->Neighbours}
      [2, Daniel]->{1->Friends}
      [3, Edward]->{2->Neighbours}
  All messages:
      1->Hi , How are you doing?
      2->I like pizza!
      3-> WOOO , I like this
  New messages:
      [1, Sheree]->{3}
      [3, Edward]->{1}
  Old/read messages:
      [2, Daniel]->{2}

->list_new_messages(2)
  27:  OK
  There are no new messages for this user.

->list_old_messages(2)
  28:  OK
  Old/read messages for user [2, Daniel]:
      2->I like pizza!

->read_message(3,2)
  29:  ERROR 
  User not authorized to access this message.

->list_new_messages(3)
  30:  OK
  New/unread messages for user [3, Edward]:
      1->Hi , How are you doing?

->list_old_messages(3)
  31:  OK
  There are no old messages for this user.

->read_message(1,3)
  32:  OK
  Message for user [1, Sheree]: [3, " WOOO , I like this"]
  Users:
      1->Sheree
      2->Daniel
      3->Edward
  Groups:
      1->Friends
      2->Neighbours
  Registrations:
      [1, Sheree]->{1->Friends, 2->Neighbours}
      [2, Daniel]->{1->Friends}
      [3, Edward]->{2->Neighbours}
  All messages:
      1->Hi , How are you doing?
      2->I like pizza!
      3-> WOOO , I like this
  New messages:
      [3, Edward]->{1}
  Old/read messages:
      [1, Sheree]->{3}
      [2, Daniel]->{2}

->list_new_messages(1)
  33:  OK
  There are no new messages for this user.

->list_old_messages(1)
  34:  OK
  Old/read messages for user [1, Sheree]:
      3-> WOOO , I like this

->delete_message(2,1)
  35:  ERROR 
  Message with this ID not found in old/read messages.

->delete_message(2,2)
  36:  OK
  Users:
      1->Sheree
      2->Daniel
      3->Edward
  Groups:
      1->Friends
      2->Neighbours
  Registrations:
      [1, Sheree]->{1->Friends, 2->Neighbours}
      [2, Daniel]->{1->Friends}
      [3, Edward]->{2->Neighbours}
  All messages:
      1->Hi , How are you doing?
      2->I like pizza!
      3-> WOOO , I like this
  New messages:
      [3, Edward]->{1}
  Old/read messages:
      [1, Sheree]->{3}

->delete_message(2,3)
  37:  ERROR 
  Message with this ID not found in old/read messages.

->list_old_messages(2)
  38:  OK
  There are no old messages for this user.

->list_new_messages(3)
  39:  OK
  New/unread messages for user [3, Edward]:
      1->Hi , How are you doing?
